File: designPatterns/Behavioral_patterns/Strategy/python/ConcreteStrategyA.py
from Strategy import Strategy


class ConcreteStrategyA(Strategy):
    """
    Concrete Strategies implement the algorithm while following the base Strategy
    interface. The interface makes them interchangeable in the Context.
    """

    def do_algorithm(self, data):
        return sorted(data)


class ConcreteStrategyB(Strategy):
    def do_algorithm(self, data):
        return reversed(sorted(data))

File: designPatterns/Behavioral_patterns/Strategy/typescript/ConcreteStrategyA.ts
import Strategy from "./Strategy";

/**
 * Concrete Strategies implement the algorithm while following the base Strategy
 * interface. The interface makes them interchangeable in the Context.
 */
export default class ConcreteStrategyA implements Strategy {
  public doAlgorithm(data: string[]): string[] {
    return data.sort();
  }
}
